crates/wascc-provider/src/lib.rs

```diff
@@ -133,33 +148,28 @@ impl<S: ModuleStore + Send + Sync> WasccProvider<S> {
         //
         // Here we are using the native capabilties as statically linked libraries that will
         // be compiled into the wascc-provider binary.
-        let cloned_host = host.clone();
-        tokio::task::spawn_blocking(move || {
-            info!("Loading HTTP Capability");
-            let http_provider = HttpServerProvider::new();
-            let data = NativeCapability::from_instance(http_provider, None)
-                .map_err(|e| anyhow::anyhow!("Failed to instantiate HTTP capability: {}", e))?;
+        info!("Loading HTTP capability");
+        let http_provider = HttpServerProvider::new();
+        let data = NativeCapability::from_instance(http_provider, None)
+            .map_err(|e| anyhow::anyhow!("Failed to instantiate HTTP capability: {}", e))?;
+        host.lock()
```

Comment:
The host lock should use a `tokio::sync::Mutex` if we are going to take this out of the `spawn_blocking`

Reply:
once again, my blind copy/paste.... The old code cloned the host lock and threw it in a spawn_blocking. Let me fix that up.